### PR TITLE
test(IA-4623): add tests for AzureMonitorLinuxAgent retry logic

### PR DESCRIPTION
**Pull request by cpate4**
Also updates README with required `jq` installation.

### CODE REVIEW

File: service/src/test/java/bio/terra/workspace/service/resource/controlled/cloud/azure/vm/EnableVmLoggingStepTest.java [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+          @Override
+          void createDataCollectionRuleAssociation(
+              FlightContext context, ApiAzureLandingZoneDeployedResource dcr, String vmId) {
+            // this has to do with creating the extension ...
+            // and we aren't validating that here
+          }
+        };
+
+    assertEquals(step.doStep(flightContext), StepResult.getStepResultSuccess());
+    verify(updateMachineMock, times(1)).withoutExtension("AzureMonitorLinuxAgent");
```
Comment (aherbst-broad) [inline service/src/test/java/bio/terra/workspace/service/resource/controlled/cloud/azure/vm/EnableVmLoggingStepTest.java:162]: nit: times(1) is the [default](https://javadoc.io/doc/org.mockito/mockito-core/latest/org/mockito/Mockito.html#verify(T)) and can be omitted.

File: service/src/test/java/bio/terra/workspace/service/resource/controlled/cloud/azure/vm/EnableVmLoggingStepTest.java [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+  @Mock private HttpResponse mockHttpResponse;
+
+  @Mock private FlightContext flightContext;
+
+  private BearerToken bearerToken;
+  private VirtualMachine.Update updateMachineMock;
+
+  @BeforeEach
+  void setup() {
```
Comment (aherbst-broad) [inline service/src/test/java/bio/terra/workspace/service/resource/controlled/cloud/azure/vm/EnableVmLoggingStepTest.java:56]: Out of scope for this review, but food for thought: I'm starting to push more on separating this sort of cloud business logic out of the "raw" stairway steps into helpers/utils/services/whatever to ease testing. 

Ideally we'd be able to isolate the azure interaction concerns and expose a saner interface to the stairway orchestration so we can exercise them separately in tests (or integration tests) and don't have to go into the heavyweight mocking as much.

**Review by aherbst-broad — APPROVED** [review]
lgtm, thanks for adding the coverage.

File: service/src/test/java/bio/terra/workspace/service/resource/controlled/cloud/azure/vm/EnableVmLoggingStepTest.java (hunk 2 above)
Comment (cpate4) [reply service/src/test/java/bio/terra/workspace/service/resource/controlled/cloud/azure/vm/EnableVmLoggingStepTest.java:56]: 😬 ... this was definitely the pain point of getting this in. I don't mind moving it over there to help future people in this repo! ... here, hold my (insert-beverage-here)

Comment (cpate4) [reply service/src/test/java/bio/terra/workspace/service/resource/controlled/cloud/azure/vm/EnableVmLoggingStepTest.java:56]: I was hoping to do this before break ... but got interrupted by other things. 
I am going to:
- get this merged, and
- follow-up with you to figure out how to do this (refactor)

Comment (dvoet) [reply service/src/test/java/bio/terra/workspace/service/resource/controlled/cloud/azure/vm/EnableVmLoggingStepTest.java:56]: I don't have enough thumbs to express my support for this

File: service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/vm/EnableVmLoggingStep.java [hunk 3]
```diff
@@ -78,7 +79,7 @@ public StepResult doStep(FlightContext context) throws InterruptedException, Ret
   }
 
   @WithSpan
-  private void createDataCollectionRuleAssociation(
+  void createDataCollectionRuleAssociation(
```
Comment (dvoet) [inline service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/vm/EnableVmLoggingStep.java:82]: add `@VisibleForTesting` annotation to this as well as `createExtension` and `removeExtension`

**Review by dvoet — APPROVED** [review]
1 minor comment

File: service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/vm/EnableVmLoggingStep.java (hunk 3 above)
Comment (cpate4) [reply service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/vm/EnableVmLoggingStep.java:84]: Added `com.google.common.annotations.VisibleForTesting` to these three methods. Thanks!

**Pull request merged**